src/entity/CommonGame.java

```diff
@@ -0,0 +1,102 @@
+package entity;
+
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.UUID;
+
+public class CommonGame implements Game {
+    private final String ID;
+    private int score;
+    private final String difficulty;
+    private final String genre;
+    private final int initialLives;
+    private final int maxRounds;
+    private int currentLives;
+    private int roundsPlayed = 0;
+    private ArrayList<Round> allRounds;
+    private final LocalDateTime createdAt;
+    private LocalDateTime finishedAt;
+
+    CommonGame(String genre, String difficulty, int maxRounds,
+               int initialLives) {
+        this.ID = UUID.randomUUID().toString();
+        this.genre = genre;
+        this.difficulty = difficulty;
+        this.maxRounds = maxRounds;
+        this.initialLives = initialLives;
+        this.createdAt = LocalDateTime.now();
+    }
+
+    @Override
+    public String getID() {
+        return ID;
+    }
+
+    @Override
+    public String getDifficulty() {
+        return difficulty;
+    }
+
+    @Override
+    public String getGenre() {
+        return genre;
+    }
+
+    @Override
+    public int getInitialLives() {
+        return initialLives;
+    }
+
+    @Override
+    public int getMaxRounds() {return maxRounds;}
+
+    @Override
+    public int getCurrentLives() {
+        return currentLives;
+    }
+
+    @Override
+    public int getRoundsPlayed() {
+        return roundsPlayed;
+    }
```

Comment:
Calling `setCurrentRound` and then `getRoundsPlayed` right now leads to unexpected results (i.e. returns 0 rather than 1) since `roundsPlayed` never gets updated anywhere

Reply:
Right, I hadn't full implemented that yet as I wasn't sure how we were going to update it, did you want to do it by the size of the list `allRounds`?

Reply:
Gotcha. Sure, that would work! 👍